magento/pwa-studio
# [PWA-2012]: Update Apollo Client

## Description

This PR updates `@apollo/client` to [latest version](https://github.com/apollographql/apollo-client/releases/tag/v3.4.0). This update brings better performance and fixes one of the problems we were solving manually with `@connection` directive, when Apollo would cache customer PII on `cache.ROOT_MUTATION`.
 
The following changes were made due to the update:
- Removed all unnecessary  `@connection(key: *)` directives from mutations.
-  Included `__DEV__` variable on Webpack DefinePlugin.

## Related Issue







Closes PWA-2012

## Acceptance






### Verification Stakeholders



### Specification




## Verification Steps







#### Test scenario(s) for direct fix/feature





1. Verify installed `@apollo/client` is latest version
3. Verify if customer PII information is not being stored on `ROOT_MUTATION` cache.
2. Overall smoke test to verify no regressions of items in "potentially disruptive changes" section of the Apollo Client release notes.

#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
